cinema/views.py

```diff
@@ -56,3 +89,66 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                        - Count("tickets")
+                    )
+                )
+            )
+
+        if self.action == "retrieve":
+            queryset = queryset.select_related("cinema_hall", "movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            movie = _params_to_ints(movie)
```

Comment:
In `MovieSessionViewSet`, before calling `_params_to_ints(movie)`, ensure that `movie` is not `None` to prevent potential errors.